misc/services/camel/de-metas-camel-externalsystems/de-metas-camel-metasfresh/src/main/java/de/metas/camel/externalsystems/metasfresh/restapi/MassJsonRequestDispatcherRoute.java

```diff
@@ -0,0 +1,226 @@
+		}
+	}
+	
+	private void moveParserCursorInsideItemBody(@NonNull final JsonParser parser)
+	{
+		try
+		{
+			parser.nextToken();
+			final String fieldName = parser.getCurrentName();
+			if (!fieldName.equals(JsonMassUpsertRequest.FIELD_NAME_ITEM_BODY))
+			{
+				throw new RuntimeCamelException("Error parsing the request: next fieldName shall be " + JsonMassUpsertRequest.FIELD_NAME_ITEM_BODY);
+			}
+		}
+		catch (final Exception exception)
+		{
+			throw new RuntimeCamelException("Could not parse the JSON!" + exception.getMessage());
+		}
+	}
+
+	@NonNull
+	private String getItemType(@NonNull final JsonParser parser)
+	{
+		try
+		{
+			parser.nextToken();
+			final String fieldName = parser.getCurrentName();
+			if (fieldName.equals(JsonMassUpsertRequest.FIELD_NAME_ITEM_TYPE))
+			{
+				parser.nextToken();
+
+				return jsonMapper.readValue(parser, String.class);
+			}
+			else
+			{
+				throw new RuntimeCamelException("Error parsing the request: next fieldName shall be " + JsonMassUpsertRequest.FIELD_NAME_ITEM_TYPE);
+			}
+		}
+		catch (final Exception exception)
+		{
+			throw new RuntimeCamelException("Could not parse the JSON!" + exception.getMessage());
+		}
+	}
+
+	@NonNull
+	private String getBatchId(@NonNull final JsonParser parser)
+	{
+		try
+		{
+			parser.nextToken();
+			final String fieldName = parser.getCurrentName();
+			if (fieldName.equals(JsonMassUpsertRequest.FIELD_NAME_BATCH_ID))
+			{
+				parser.nextToken();
+				return jsonMapper.readValue(parser, String.class);
+			}
+			else
+			{
+				throw new RuntimeCamelException("Error parsing the request: next fieldName shall be " + JsonMassUpsertRequest.FIELD_NAME_BATCH_ID);
+			}
+		}
+		catch (final Exception exception)
+		{
+			throw new RuntimeCamelException("Could not parse the JSON!" + exception.getMessage());
+		}
+	}
+
+	private void registerMassUploadStatisticsCollector(
+			@NonNull final Exchange exchange,
+			@NonNull final String batchId)
+	{
+		final String externalSystemConfigValue = FilenameUtil.getExternalSystemConfigValue(exchange.getIn().getHeader(FILE_NAME_HEADER, String.class));
+		final FeedbackConfig feedbackConfig = feedbackConfigProvider.getFeedbackConfig(externalSystemConfigValue).orElse(null);
+
+		final MassUpsertStatisticsCollector massUpsertStatisticsCollector = MassUpsertStatisticsCollector.builder()
+				.batchId(batchId)
+				.feedbackConfig(feedbackConfig)
+				.build();
+
+		exchange.setProperty(MASS_UPLOAD_STATISTICS_COLLECTOR_PROPERTY, massUpsertStatisticsCollector);
+	}
+	
+	private static void validateRootIsObject(@NonNull final JsonParser parser)
+	{
+		try
+		{
+			if (parser.nextToken() != JsonToken.START_OBJECT)
+			{
+				parser.close();
+				throw new RuntimeCamelException("Error parsing the request: root should be object");
+			}
+		}
+		catch (final Exception exception)
+		{
+			throw new RuntimeCamelException("Could not parse the JSON!" + exception.getMessage());
```

Comment:
why not add the actual root cause? doesn't RuntimeCamelException support this?
Note that if you know that only a RuntimeCamelException might be thrown, then only catch a RuntimeCamelException